exporters/otlp/otlplog/otlploghttp/exporter.go

```diff
@@ -42,8 +46,15 @@ func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
 // Shutdown shuts down the Exporter. Calls to Export or ForceFlush will perform
 // no operation after this is called.
 func (e *Exporter) Shutdown(ctx context.Context) error {
-	// TODO: implement.
-	return nil
+	var err error
+	e.shutdownOnce.Do(func() {
+		e.clientMu.Lock()
+		client := e.client
+		e.client = shutdownClient{}
```

Comment:
Cannot we  make an atomic store of shutdownClient here and get rid of sync.Once and mutex?
And we can always return nil.
The client interface would need only a `UploadLogs` method.

Also wouldn't a simple atomic.Bool would be good enough?

Reply:
The HTTP client will need a shutdown method. I don't think `client` can be defined with just `UploadLogs`.

Reply:
> The HTTP client will need a shutdown method

What it will do?

Reply:
> Cannot we make an atomic store of shutdownClient here and get rid of sync.Once and mutex?

`client` is an interface so it cannot be stored in an `atomic.Value` directly. The different implementations will cause a panic.

I can look at refactoring to use atomics though.

Reply:
> > The HTTP client will need a shutdown method
> 
> What will it do?

I expect similar to existing:

https://github.com/open-telemetry/opentelemetry-go/blob/66284071defe2c6b2261aecfa506f37d6e2a31f2/exporters/otlp/otlpmetric/otlpmetrichttp/client.go#L108-L114

Reply:
The batch processor never calls Export nor ForceFlush after Shutdown, right? Maybe we could do the same in simple processor and simplify the invariants (requirements) for exporter implementations?

Reply:
Updated. PTAL

Reply:
That was my thought as well.

Reply:
OK, waiting for the comment 😉